### PR TITLE
[SURGE-173] Curated Products

### PR DESCRIPTION
**Pull request by jordanpagewhite**
This mostly implements the Curated Products per the DX Surge UX designs.

curated_products1
curated_products2
curated_products3

#### Questions

- What is the purpose of the Four up and Basic visual styles now in this new
design? @gdoyle1 and @sjcox-rh

### CODE REVIEW

**Comment by mindreeper2420** [comment]
@jordanpagewhite 

> @mindreeper2420 Do you have any insight into these questions?
> 
>     * How can we center cards within a pf-grid or pf-gallery?
* Gallery layouts are not meant to have 'centered' content. They are to be used when you want content (such as cards) to be the same size and evenly spaced.
* Grid layouts can have centered content only through calculating the offset+column width. Some of the examples are centered by having the content by 6 columns with an offset of 3.
* If you want to center content in a container, the best bet would be to use the `pf-l-flex` layout. You can then use the justify and align properties to handle the x and y axis.
>     * How can we override the padding-top and padding-bottom styles
>       applied via pf-u-py-xl with an !important?
* You would have to apply `pf-u-py-0` after that class is applied. The second `!important` from the `py-0` would overwrite the first one.
>

File: src/styles/rhd-theme/components/_curated-products.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    border-top: 6px solid var(--rhd-theme--color--ui-accent);
+  }
+  &.basic .rhd-c-card {
+    border-top: none;
+  }
+  &.dark {
+    background: #242424;
+    
+    & .featured_products__title {
+      color: #ffffff;
```
Comment (mindreeper2420) [inline src/styles/rhd-theme/components/_curated-products.scss]: replace `#ffffff;` with `var(--pf-global--Color--light-100);`

File: src/styles/rhd-theme/components/_curated-products.scss [hunk 2]
```diff
@@ -0,0 +1,29 @@
+.assembly-type-featured_products {
+  background-color: var(--rhd-gray-2);
+
+  .featured_products__title {
+    text-align: center;
+  }
+
+  .featured_products__cards {
```
Comment (mindreeper2420) [inline src/styles/rhd-theme/components/_curated-products.scss]: This can be deleted, as it is empty.

**Comment by gdoyle1** [comment]
> * What is the purpose of the Four up visual style now in this new
>   design?
> * Are we using the No padding top and No padding bottom visual styles?

@jordanpagewhite @sjcox-rh The purpose of the four visual style (which I believe should now be the standard) is that it now allows us to display more content per row and gives the site a denser look than beforehand. All of the top and bottom padding should be 24px across assemblies but yes we should still provide the no padding top and no padding bottom visual styles

**Comment by jordanpagewhite** [comment]
@mindreeper2420 Regarding the image markup: That markup is generated by Drupal, and I haven't yet figured out a _good_ way to get the `.rhd-c-card__image` class in there. I am reluctant to just entirely override what Drupal is doing there, as we have (or will) with other component because in the instance of images, Drupal applies image formatters. This will likely require writing some PHP so that we can preserve this Drupal functionality _and_ customize the markup.

**Comment by jordanpagewhite** [comment]
Thank you SJ and Gina. I am addressing that feedback now. I also noticed a visual bug that impacts all `.rhd-c-card` components, and I created an issue here to track it https://github.com/redhat-developer/rhd-frontend/issues/229

**Comment by jordanpagewhite** [comment]
Updated Product cards with spacing around img element:

localhost_1313_components_curated-products_ (3)
localhost_1313_components_curated-products_ (2)
localhost_1313_components_curated-products_ (1)

Sorry my full page fancy screenshots aren't working for some reason in Chrome :( I will have to figure out how to resolve that soon. I love the full page screenshots

**Comment by sjcox-rh** [comment]
@jordanpagewhite much better, UX approved!

**Pull request merged**